### PR TITLE
Homepage branch

### PR DESCRIPTION
**Pull request by nayyyhaa**
## In this PR :
- I've just added Homepage layout to my video library.

**ONLY check the following files**

#### Pages: 
- [x] [pages/AllVideosFeed.js](https://github.com/nayyyhaa/inNotion-watch/compare/homepage-branch?expand=1#diff-b96bc22ca50f509524073a3a2f33c017bab1904a9d981de01e0b1b7575448261)
- [x] [pages/Homepage.js](https://github.com/nayyyhaa/inNotion-watch/compare/homepage-branch?expand=1#diff-1063f1bef92d5c0fee5f64413fed09584b31b2d9ab0873a1166a7a2f8294aa4c)

#### Component files

- [x] [/components/Navbar.jsx](https://github.com/nayyyhaa/inNotion-watch/compare/homepage-branch?expand=1#diff-0b9e115b879d5389a7caf5029cd442d7358dbdeffd27469a478ad90752ed3d54)
- [x] [components/Sidebar.jsx](https://github.com/nayyyhaa/inNotion-watch/compare/homepage-branch?expand=1#diff-0f7804eafe5f56106581a7149f16bcf316a4466155780612661ed7bffb305fb1)
- [x] [components/Video.jsx](https://github.com/nayyyhaa/inNotion-watch/compare/homepage-branch?expand=1#diff-90cebbf5c9dabb90902386473612919c621846cfc5d628545fee99328256661c)

 #### Contexts  + Reducers

- [x] [contexts/SidebarContext.js](https://github.com/nayyyhaa/inNotion-watch/compare/homepage-branch?expand=1#diff-6d263c696159af1acb455b8fc974d0320e2422015759266c510d8a4bd6766a63)

#### files to *ignore*
- [ ] data files
- [ ] CSS files
- [ ] Backend files

### CODE REVIEW

File: src/components/Chip.jsx [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const Chip = ({ category }) => {
+  return <div class="chip m-v-2 m-h-1 cursor">{category}</div>;
```
Comment (govindappaarun) [inline src/components/Chip.jsx]: It should be className

File: src/components/Navbar.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { useSidebar } from "contexts/SidebarContext";
+
+export const Navbar = () => {
+  const { setShowFilterBar } = useSidebar();
+  return (
+    <>
```
Comment (govindappaarun) [inline src/components/Navbar.jsx]: this fragment is not required

File: src/components/Sidebar.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { useSidebar } from "contexts/SidebarContext";
+
+export const Sidebar = () => {
+  const { showFilterBar, setShowFilterBar } = useSidebar();
+  const location = useLocation();
+  return (
+    <div className={`side-bar video-sidebar w-20p ${showFilterBar ? "show-filter" : ""}`}>
+      <ul className="sidebar-items no-bullet col-flex flex-start no-wrap p-h-2 m-v-3">
+        {sidebarData?.map(({ _id, title, link, icon }) => {
+          const Icon = icon;
```
Comment (govindappaarun) [inline src/components/Sidebar.jsx]: suggestion: you can use icon: Icon in the destructing itself

File: src/components/index.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export { Navbar } from "./Navbar";
```
Comment (govindappaarun) [inline src/components/index.js:1]: suggestion: you can short this line by `export * from  "./Navbar" ` it exports everything from that file

File: src/contexts/SidebarContext.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { createContext, useState, useContext } from "react";
+
+const SidebarContext = createContext();
+
+const SidebarProvider = ({ children }) => {
+  const [showFilterBar, setShowFilterBar] = useState(false);
+
+  return <SidebarContext.Provider value={{ showFilterBar, setShowFilterBar }}>{children}</SidebarContext.Provider>;
```
Comment (govindappaarun) [inline src/contexts/SidebarContext.js:8]: suggestion: you are missing code format like prettier, configure in vscode to run on save of files or before commit

File: src/pages/Homepage.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+export const Homepage = () => {
+  const location = useLocation();
+  return (
+    <>
+      <div className="video-layout row-flex no-wrap flex-start">
+        <Sidebar />
+        <main className="videos-content">
+          {location.pathname === "/" && <AllVideosFeed />}
+
+          {/* NOTE: Will be uncommenting after adding pages */}
```
Comment (govindappaarun) [inline src/pages/Homepage.js:14]: suggestion: if these lines are not required, you can remove them

**Review by govindappaarun — APPROVED** [review]
Looks good to merge, apart from few minor improvements
Good separation of code into components and directory structure

File: src/components/Chip.jsx (hunk 1 above)
Comment (nayyyhaa) [reply src/components/Chip.jsx]: Done with change.

File: src/components/Sidebar.jsx (hunk 3 above)
Comment (nayyyhaa) [reply src/components/Sidebar.jsx]: Done with change.

File: src/components/index.js (hunk 4 above)
Comment (nayyyhaa) [reply src/components/index.js:1]: Sure will make this change.

File: src/contexts/SidebarContext.js (hunk 5 above)
Comment (nayyyhaa) [reply src/contexts/SidebarContext.js:8]: I've saved this file with prettier, it's formatting like this only.

File: src/pages/Homepage.js (hunk 6 above)
Comment (nayyyhaa) [reply src/pages/Homepage.js:14]: Sure will remove them after adding the required pages.

File: src/components/Navbar.jsx (hunk 2 above)
Comment (nayyyhaa) [reply src/components/Navbar.jsx]: Done with change.

**Review by ApurvaSawant11 — APPROVED** [review]
Looks good to me! Just make the changes suggested by Arun

**Comment by nayyyhaa** [comment]
Done with the changes. Thanks @ApurvaSawant11 & @govindappaarun for reviewing.

**Pull request merged**